src/main/java/org/gridsuite/study/server/service/LoadFlowService.java

```diff
@@ -0,0 +1,234 @@
+import java.util.stream.Collectors;
+
+import static org.gridsuite.study.server.StudyConstants.*;
+import static org.gridsuite.study.server.StudyException.Type.*;
+
+/**
+ * @author Kevin Le Saulnier <kevin.lesaulnier at rte-france.com>
+ */
+@Service
+public class LoadFlowService {
+
+    static final String RESULT_UUID = "resultUuid";
+    private String loadFlowServerBaseUri;
+
+    @Autowired
+    NotificationService notificationService;
+
+    private final NetworkService networkStoreService;
+
+    NetworkModificationTreeService networkModificationTreeService;
+    private final ObjectMapper objectMapper;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Autowired
+    public LoadFlowService(
+            @Value("${gridsuite.services.loadflow-server.base-uri:http://loadflow-server/}") String loadFlowServerBaseUri,
+            NetworkModificationTreeService networkModificationTreeService,
+            NetworkService networkStoreService, ObjectMapper objectMapper) {
+        this.loadFlowServerBaseUri = loadFlowServerBaseUri;
+        this.networkStoreService = networkStoreService;
+        this.networkModificationTreeService = networkModificationTreeService;
+        this.objectMapper = objectMapper;
+    }
+
+    public UUID runLoadFlow(UUID studyUuid, UUID nodeUuid, LoadFlowParametersInfos loadflowParameters, String provider, String userId) {
+        UUID networkUuid = networkStoreService.getNetworkUuid(studyUuid);
+        String variantId = getVariantId(nodeUuid);
+        UUID reportUuid = getReportUuid(nodeUuid);
+
+        String receiver;
+        try {
+            receiver = URLEncoder.encode(objectMapper.writeValueAsString(new NodeReceiver(nodeUuid)), StandardCharsets.UTF_8);
+        } catch (JsonProcessingException e) {
+            throw new UncheckedIOException(e);
+        }
+
+        var uriComponentsBuilder = UriComponentsBuilder
+                .fromPath(DELIMITER + LOADFLOW_API_VERSION + "/networks/{networkUuid}/run-and-save")
+                .queryParam(QUERY_PARAM_RECEIVER, receiver)
+                .queryParam("reportUuid", reportUuid.toString())
+                .queryParam("reporterId", nodeUuid.toString());
+        if (!provider.isEmpty()) {
+            uriComponentsBuilder.queryParam("provider", provider);
+        }
+        if (!StringUtils.isBlank(variantId)) {
+            uriComponentsBuilder.queryParam(QUERY_PARAM_VARIANT_ID, variantId);
+        }
+        var path = uriComponentsBuilder.buildAndExpand(networkUuid).toUriString();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set(HEADER_USER_ID, userId);
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        HttpEntity<LoadFlowParametersInfos> httpEntity = new HttpEntity<>(loadflowParameters, headers);
+
+        return restTemplate.exchange(loadFlowServerBaseUri + path, HttpMethod.POST, httpEntity, UUID.class).getBody();
+    }
+
+    public void deleteLoadFlowResult(UUID uuid) {
```

Comment:
This method should be called : 
- when the study is deleted 
- when the node is deleted or invalidated
- when the LF parameters are modified
You can look where is called deleteSensitivityAnalysisResult()

Reply:
done